src/components/Search/CompareWithBase.tsx

```diff
@@ -161,6 +155,34 @@ function CompareWithBase({
     setInProgressNewRevs(newStagingRevs);
   };
 
+  const handleItemToggleInChangesetList = ({
```

Comment:
hey carla, sorry if I wasn't clear in my comment.

What I was refering to "overkill" was specifically the use of a hook. But I started the comment with :

> Extracting handleItemToggleInChangesetList and handleRemoveNewRevision is a good idea because they're quite complex functions and duplicating them doesn't feel right.

My suggestion was to extract them to a separate file as normal function.

But duplicating them is not too bad either.
Some people say that duplicating only twice is OK, and only from 3 times they start to think about reducing the duplication.
"The rule of three" => https://stackoverflow.com/a/2298357/5169882 
The rationale is that (as I said also in other occasions) keeping the related code in the same file makes it easier to read, while moving it elsewhere requires more hops to understand what's going on (hence my focus on naming things correctly if we do that).

So.... bottom line is, let's keep it that way. But if we ever need to copy paste this again, we'll move them out to another place.

Reply:
Okay, I hear you. Yes, let's leave it for now and I'll extract to a new file within the Search folder if it comes up again.